src/functions/__tests__/casing.test.ts

```diff
@@ -1,26 +1,27 @@
-import { casing, ICasingOptions } from '../casing';
+import { AssertionError } from 'assert';
+import { casing, CasingType } from '../casing';
 
-function runCasing(target: unknown, type: ICasingOptions['type'], disallowDigits?: boolean) {
+function runCasing(target: unknown, type: CasingType, disallowDigits?: boolean, separator?: any) {
```

Comment:
Note: separator is typed as `any` on purpose in order to be able to inject anything (even if that doesn't match the expected TyperScript type declaration), in order to assess the function behavior in corner cases scenarios.